dali/operators/reader/loader/video/frames_decoder_gpu.h

```diff
@@ -58,6 +58,14 @@ class DLL_PUBLIC FramesDecoderGpu : public FramesDecoder {
    */
   explicit FramesDecoderGpu(const std::string &filename, cudaStream_t stream = 0);
 
+  /**
+ * @brief Construct a new FramesDecoder object.
+ * 
+ * @param memory_file Pointer to memory with video file data.
+ * @param memory_file_size Size of memory_file.
+ */
+  explicit FramesDecoderGpu(char *memory_file, int memory_file_size, cudaStream_t stream = 0);
```

Comment:
How about using `span` instead of passing the pointer and size separately?

Reply:
`span` is C++ 20

Reply:
We actually do have our own implementation of `span`, so you could use that one. But it's up to you.

Reply:
This code is not dependent on DALI other then some error checking and reporting and I think it would be good to keep it this way. It might be useful to others, I think.